pycaret/internal/plots/time_series.py

```diff
@@ -791,3 +832,106 @@ def plot_predictions_with_confidence(
         return fig, data, predictions, upper_interval, lower_interval
     else:
         return fig
+
+
+def plot_time_series_decomposition(
+    data: pd.Series,
+    trend_component:pd.Series,
```

Comment:
I think the`plot_time_series_decomposition` should only take data as argument. The decomposition should be done inside that function. You may choose to take a `data_kwargs` with key = `type` and values = `classical` or `stl`. Then based on that do the appropriate decomposition inside that function.

User can call from top as `plot_model(plot='decomp_classical')` or `plot_model(plot='decomp_stl')`. In that case I can pass the right key from outside to your function so end user does not have to worry about it.